src/optimistic-oracle/optimistic-oracle.js

```diff
@@ -0,0 +1,232 @@
+
+    // process the target events
+    /* eslint-disable no-await-in-loop */
+    for (let i = 0; i < parsedLogs.length; i++) {
+      const log = parsedLogs[i];
+      if (log.name === 'RequestPrice') {
+        // get token address (this will be checksum encoded)
+        const { currency, requester } = log.args;
+
+        // make a request to the price feed API
+        let price;
+        try {
+          price = await getPrice(currency);
+        } catch (error) {
+          console.error(error);
+          continue;
+        }
+
+        // report the price obtained as a finding
+        findings.push(Finding.fromObject({
+          name: 'UMA Price Request',
+          description: `Price requested from Optimistic Oracle.  Currency=${currency}, Price=${price}`,
+          alertId: 'AE-UMA-OO-REQUESTPRICE',
+          severity: FindingSeverity.Low,
+          type: FindingType.Unknown,
+          protocol: 'uma',
+          everestId: config.umaEverestId,
+          metadata: {
+            requester,
+            currency,
+            price,
+          },
+        }));
+      }
+
+      if (log.name === 'ProposePrice') {
+        const { currency } = log.args;
+        const { requester } = log.args;
+        const { proposer } = log.args;
+
+        // set up ERC-20 contract to get decimals value
+        // in production, erc20Contract will always be undefined
+        // in testing, erc20Contract will be assigned to a mock contract
+        /* eslint-disable no-param-reassign */
+        if (erc20Contract === undefined) {
+          erc20Contract = new ethers.Contract(currency, erc20Abi, provider);
+        }
+        /* eslint-enable no-param-reassign */
+
+        const decimals = await erc20Contract.decimals();
+
+        // convert proposedPrice to a human-readable decimal value
+        let proposedPrice = ethers.utils.formatUnits(log.args.proposedPrice, decimals);
+
+        // convert the proposedPrice to a BigNumber type for difference calculations later
+        proposedPrice = new BigNumber(proposedPrice.toString());
+
+        // make a request to the price feed API
+        // CoinGecko will return decimal values
+        let price;
+        try {
+          price = await getPrice(currency);
+        } catch (error) {
+          console.error(error);
+          continue;
+        }
+
+        // convert the price obtained to a BigNumber type
+        price = new BigNumber(price);
+
+        // generate a finding if the price difference threshold (defined in agent-config.json) has
+        // been exceeded
+        const percentError = calculatePercentError(proposedPrice, price);
+
+        if (percentError.isGreaterThan(config.optimisticOracle.priceThresholdPct)) {
+          findings.push(Finding.fromObject({
+            name: 'UMA Price Proposal',
+            description: `Price proposed to Optimistic Oracle is disputable. Currency=${currency}, ProposedPrice=${proposedPrice}, Price=${price}`,
```

Comment:
I don't think the price is always in USD, reallllllllllly not sure though. I've been looking around their code and I think it's per contract and based off of the price in terms of collateral token, I'll look into this more

the reason I think it's not always in USD is because their price feeds for coin gecko can be accepted in anything, not just USD

Reply:
it would be in a config like how they build theirs. the thing isn't that it's possible to get that info it's more like if the on chain contract is expecting something to be valued in something other than USD we can't specify that. BUT I'm not 100% sure there is more than just USD being quoted.